src/lib.rs

```diff
@@ -75,8 +75,7 @@ pub struct Cactus<T> {
 #[derive(Clone)]
 struct Node<T> {
     val: T,
-    parent: Option<Rc<Node<T>>>,
-    len: usize
+    parent: Option<Rc<Node<T>>>
```

Comment:
I would have left the trailing comma on. a) less effort b) easier to review.

Reply:
The trailing comma troubles me. I think it's because I suspect it must make things harder to parse...

Reply:
Each to their own I guess, but I swear by them.